### PR TITLE
Add support for moving pieces

### PR DESCRIPTION
**Pull request by dynnog**
Done:
- Add pieces' path to routes.rb file.
- Add pieces controller and pass piece's ID into pieces.
- Correctly redirect to new page with the correct pieces ID.
- Highlight the chosen piece.
- Change every squares into game_path links.
- Add circle icons on empty squares.
- Update the piece x_position and y_position of the new square.
- Redirect back to game_path.
- Install database_cleaner gem.

Not done:

### CODE REVIEW

File: app/views/games/show.html.erb [hunk 1]
```diff
@@ -1,14 +1,16 @@
...
 <table class="chess_table">
   <% 7.downto(0).each do |row_number| %>
     <tr>
       <% 0.upto(7).each do |column_number| %>
         <td class="<%= (row_number + column_number).even? ? "black_space" : "white_space"%>">
-          <% piece = @game.pieces.find_by(x_position: column_number, y_position: row_number) %>
+          <% piece = current_game.pieces.find_by(x_position: column_number, y_position: row_number) %>
           <% if piece %>
-            <p class='piece'><%= piece.to_unicode %></p>
+            <%= link_to game_piece_path(current_game.pieces) do %>
```
Comment (mscoutermarsh) [inline app/views/games/show.html.erb]: Try changing this to `game_piece_path(game, piece)` (I think that's correct).

In a console, if you run `rake routes`. You'll be able to see all of the routes and what their `helper` is. So you should see a `game_piece_path` there. Seeing that can sometimes be helpful in figuring out which path helper to use (all of the `something_path` methods are considered a "path helper").

😄

File: app/views/pieces/show.html.erb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+      <% 0.upto(7).each do |column_number| %>
+        <td class="<%= (row_number + column_number).even? ? "black_space" : "white_space"%>">
+          <% piece = current_game.pieces.find_by(x_position: column_number, y_position: row_number) %>
+          <% if piece && current_piece == piece %>
+            <p class='piece highlight'><%= piece.to_unicode %></p>
+          <% elsif piece %>
+            <p class='piece'><%= piece.to_unicode %></p>
+          <% end %>
+        </td>
+      <% end %>
```
Comment (mscoutermarsh) [inline app/views/pieces/show.html.erb]: Next step for this view is: Add a button that connects to `pieces#update` to move the piece.

The button can go on all the filled in squares plus the empty squares.

😄 

(Can do this in a separate pull request if you'd like)

**Review by mscoutermarsh — APPROVED** [review]

**Review by mscoutermarsh — APPROVED** [review]

File: app/controllers/pieces_controller.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+  def show
+    @piece = Piece.find(params[:id])
+  end
+
+  def update
+    row = params[:row]
+    column = params[:column]
+    piece = Piece.find(params[:id])
+    piece.move_to!(column, row)
```
Comment (mscoutermarsh) [inline app/controllers/pieces_controller.rb:16]: Looks good! Could reduce this a bit by doing the following.

```Ruby
piece = Piece.find(params[:id])
piece.move_to!(params[:column], params[:row])
```

**Review by mscoutermarsh — APPROVED** [review]

File: app/controllers/pieces_controller.rb (hunk 3 above)
Comment (dynnog) [reply app/controllers/pieces_controller.rb:16]: Thanks, I will change it for the next task!
